src/Http/Http.Extensions/gen/RequestDelegateGeneratorSources.cs

```diff
@@ -449,6 +449,27 @@ public override bool IsDefined(Type attributeType, bool inherit)
     }
 """;
 
+    public static string ParameterBindingMetadataClass = $$"""
+    {{GeneratedCodeAttribute}}
+    file sealed class ParameterBindingMetadata(
+        string name,
+        ParameterInfo parameterInfo,
+        bool hasTryParse = false,
+        bool hasBindAsync = false,
+        bool isOptional = false) : Microsoft.AspNetCore.Http.Metadata.IParameterBindingMetadata
```

Comment:
(nit)

```suggestion
        bool isOptional = false) : IParameterBindingMetadata
```

We don't fully qualify other type names in the generated code. Do we need it for this one?

Reply:
Nope -- we don't.